From 571a06f715bc12e283383a83817ba6935c3b9a5e Mon Sep 17 00:00:00 2001
From: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Date: Wed, 18 Jul 2018 15:08:01 +0300
Subject: [PATCH 4680/5242] MLK-18655-1 sound: asoc: fix regmap caching for
 micfil

commit  9be4b802d6f1bd167c2c0a79edd3cc3f2a882b53 from
https://source.codeaurora.org/external/imx/linux-imx.git

Remove regmap_cache_only from probe since this should be done
in suspend/resume or when hwvad is enabled/disabled. If regmap
is cached in probe, all volatile registers cannot be read/write
until caching is set to false (which in our case was done only
in resume function).

Signed-off-by: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
(cherry picked from commit 4cf9ddee84c995e02931be9de5448cb162130862)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_micfil.c |    2 --
 1 file changed, 2 deletions(-)

diff --git a/sound/soc/fsl/fsl_micfil.c b/sound/soc/fsl/fsl_micfil.c
index 6e96635..cf62236 100644
--- a/sound/soc/fsl/fsl_micfil.c
+++ b/sound/soc/fsl/fsl_micfil.c
@@ -779,8 +779,6 @@ static int fsl_micfil_probe(struct platform_device *pdev)
 
 	pm_runtime_enable(&pdev->dev);
 
-	regcache_cache_only(micfil->regmap, true);
-
 	ret = devm_snd_soc_register_component(&pdev->dev, &fsl_micfil_component,
 					      &fsl_micfil_dai, 1);
 	if (ret) {
-- 
1.7.9.5

